### PR TITLE
Add support in swatch-koa for client to specify exception handler

### PR DESCRIPTION
**Pull request by bwtimm**
Refactor the expose.js file to move some functions into separate files
Add support for an `onException` param in the swatch-koa options object

This is an optional function passed by the caller when setting up swatch. This function will be called inside swatch before setting an 'error' response (with ok: false). It will call the function, passing in the exception/error. The client function has the opportunity to examine the error, and either re-throw the error (so swatch will return that error) or throw a different error (which swatch will return instead), or return a value (so swatch will rescue and return that value as success).

Basically, the goal is to let a client return some generic exception whenever something bubbles up that it didnt expect, like a DB error.

### CODE REVIEW

**Comment by Furchin** [comment]
Hmmm.... I'm not sure how I feel about this. I sort of feel like it's overfitting the handler -- isn't the handler the thing that's going to be throwing the error in the first place? So one could accomplish this by simply wrapping the entire handler in a `try...catch` and translating the error into whatever error code response is desired already. I think that means we'd be going from writing

```
function errorHandler(err) {
  ... translate err into appropriate error code, eg
  if (err.message === 'Connection Timeout) {
    throw ERR_CONNECTION_TIMEOUT_MESSAGE;
  }
}

function handler(...) {
  try {
    ... some logic
  } catch (err) {
    return errorHandler(err);
  }
}

swatch(handler);
```
to

```
function errorHandler(err) {
  ... translate err into appropriate error code
  if (err.message === 'Connection Timeout) {
    throw ERR_CONNECTION_TIMEOUT_MESSAGE;
  }
}

function handler(...) {
  ... some logic
}

swatch(handler, errorHandler);
```

I guess the latter is a little bit clearer, but to me it also makes understanding how to use swatch properly more difficult since now there are more options, and it leads to potentially multiple ways to implement the same thing (since you could still do the former, so some SwatchJS consumers will do it one way, and some the other way).

**Comment by bwtimm** [comment]
So clients are going to implement one handler per endpoint/method. There will be a class of errors that can happen in every handler (generic DB errors, etc), along with an accidental runtime error because you wrote the wrong thing or accessed the wrong variable.

I want to avoid writing duplicate `try { } catch { }` code inside every handler to catch all sorts of errors. I also found it difficult to implement this kind of catch-all in the application. I tried to writing this kind of wrapper in the app and wrapping the handlers, but since swatch requires functions with matching -arity, there's no easy way to generically wrap each `handler`.

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "swatchjs-koa",
-  "version": "1.0.7",
+  "version": "1.0.8",
```
Comment (CodeFred) [inline package.json]: Should be `1.1.0` (`<breaking change>.<new feature>.<bug fix>` convention)
